lib/ansible/modules/cloud/amazon/aws_direct_connect_connection.py

```diff
@@ -0,0 +1,328 @@
+
+
+@AWSRetry.backoff(**retry_params)
+def connection_exists(client, connection_id=None, connection_name=None, verify=True):
+    try:
+        if connection_id:
+            response = client.describe_connections(connectionId=connection_id)
+        else:
+            response = client.describe_connections()
+    except botocore.exceptions.ClientError as e:
+        raise DirectConnectError(msg="Failed to describe DirectConnect ID {0}".format(connection_id),
+                                 last_traceback=traceback.format_exc(),
+                                 response=e.response)
+
+    match = []
+    connection = []
+
+    # look for matching connections
+
+    if len(response.get('connections', [])) == 1 and connection_id:
+        if response['connections'][0]['connectionState'] != 'deleted':
+            match.append(response['connections'][0]['connectionId'])
+            connection.append(response['connections'][0])
+
+    for each in response.get('connections', []):
+        if connection_name == each['connectionName'] and each['connectionState'] != 'deleted':
+            match.append(each['connectionId'])
+            connection.append(each)
+
+    # verifying if the connections exists; if true, return connection identifier, otherwise return False
+    if verify and len(match) == 1:
+        return match[0]
+    elif verify:
+        return False
+
+    # not verifying if the connection exists; just return current connection info
+    else:
+        if len(connection) == 1:
+            return {'connection': connection[0]}
+        else:
+            return {'connection': {}}
+
+
+@AWSRetry.backoff(**retry_params)
+def create_connection(client, location, bandwidth, name, lag_id):
+    if not name:
+        raise DirectConnectError(msg="Failed to create a Direct Connect connection: name required.")
+    try:
+        if lag_id:
+            connection = client.create_connection(location=location,
+                                                  bandwidth=bandwidth,
+                                                  connectionName=name,
+                                                  lagId=lag_id)
+        else:
+            connection = client.create_connection(location=location,
+                                                  bandwidth=bandwidth,
+                                                  connectionName=name)
+    except botocore.exceptions.ClientError as e:
+        raise DirectConnectError(msg="Failed to create DirectConnect connection {0}".format(name),
+                                 last_traceback=traceback.format_exc(),
+                                 response=e.response)
+    return connection['connectionId']
+
+
+def changed_properties(current_status, location, bandwidth):
+    changed = False
+
+    current_bandwidth = current_status['bandwidth']
+    current_location = current_status['location']
+
+    if (current_bandwidth != bandwidth or current_location != location):
```

Comment:
Instead of recording `changed`, you can `return (current_bandwidth != bandwidth or current_location != location)` since that conditional is what's really deciding whether to change the resource or not.